Sources/NIOHTTPCompression/HTTPDecompression.swift

```diff
@@ -134,7 +134,7 @@ extension z_stream {
     }
 
     private mutating func inflatePart(to buffer: inout ByteBuffer) throws -> Int {
-        return try buffer.writeWithUnsafeMutableBytes { pointer in
+        return try buffer.writeWithUnsafeMutableBytes(minimumWritableBytes: 0) { pointer in
```

Comment:
we reserve buffer capacity in both cases (see lines 89 in HTTPDecompression.swift and 313 in HTTPResponseCompressor.swft)

Reply:
Would it be better to refactor this to use the new API instead of capacity reservation?

Reply:
in decompressor we ensure capacity once (since we can approximately guess how much data will be written) and then write multiple times - do we really need to ensure it every time?
in compression we create a buffer with expected length from the start, so again, is there a reason to ensure capacity in this case?

Reply:
If we already have enough capacity, ensuring it costs nothing. It is no more expensive than the current code.

Reply:
done